app/scripts/controller-init/messengers/transaction-controller-messenger.ts

```diff
@@ -0,0 +1,92 @@
+import { AccountsControllerGetSelectedAccountAction } from '@metamask/accounts-controller';
```

Comment:
Hey Matt, I noticed that the messengers and controllers are in separate folders, even though they seem tightly coupled. Curious about the reasoning behind that.

Reply:
From the PR description:

>Messengers are intentionally defined in separate isolated files to allow alternate code ownership, and not over-power the controller logic and introduce bugs.

Traditionally, the messengers are defined inline with the controllers but that means any controller can casually connect or start depending on another, without any oversight or awareness from other teams.

Meaning it's a common source of bugs, and so defining them separately ensures the controller initialisation doesn't have "sudo" permissions to the entire system, but we can instead define their "permissions" elsewhere and tell each controller and initialisation instance what it's allowed to do.

Practically, it also means the platform team for example could own the messenger directories to stay informed of the high level connections between controllers.

Reply:
Thank you for the extra context!